docs/coordinates/satellites.rst

```diff
@@ -100,10 +105,12 @@ Or, if you want to find the altitude and azimuth of the satellite from a particu
 
     >>> from astropy.coordinates import EarthLocation, AltAz
     >>> siding_spring = EarthLocation.of_site('aao')  # doctest: +SKIP
-    >>> aa = teme.transform_to(AltAz(obstime=t, location=siding_spring))  # doctest: +IGNORE_WARNINGS
+    >>> topo_itrs_repr = itrs_geo.cartesian - siding_spring.get_itrs(t).cartesian  # doctest: +IGNORE_WARNINGS
```

Comment:
I had originally written my ITRS<->Observed transforms to include automatic geocentric<->topocentric ITRS transforms, which would make a direct TEME<->Observed work as likely intended, but decided that this would have complications. For instance, what of ITRS<->Observed transforms between two different `EarthLocations`? Theoretically, these would just entail a rotation about the z axis, so I could make them automatic also. Doing it this way would also require the user who actually wants the difference in stellar aberration to be included to perform an explicit ITRS<->ICRS transform.

I could still do this if you would prefer, as it would result in a "user friendly" set of self consistent transforms within ITRS. However, I sort of like the fact that Astropy currently has two branches; geocentric and topocentric, that only meet at the trunk of the tree in the ICRS. Doing the above would create a short cut between the two branches in the ITRS and require the user who wishes to use Astropy's former default behavior to do so explicitly. This is why I eventually chose to require the user to do the explicit geocentric to topocentric ITRS transform themselves and let the default remain the same in that it goes through the ICRS. 

I do think that this methodology is fairly well documented in `itrs.py` and the example transform given in Working with Earth Satellites Using Astropy Coordinates, but if you can think of ways to improve the documentation, I will be happy to include them.

Reply:
Correct me if I’m wrong (I’m getting confused and need to sit with this a bit), but doesn’t that assume the nature of the coordinate requires no handling for aberration?

eg the transform for ITRS(t) -> ITRS(t’) is different for a satellite and star?

Reply:
Yes it does. And this is another good reason to require the user to do a bit more work in setting up the topocentric ITRS frame. If I made ITRS geocentric to topocentric and topocentric to topocentric transforms automatic, then the default for these transforms within the ITRS would be to not include the effects of stellar aberration, which is contrary to the default behavior of all other Astropy coordinate transformations. 

While the vast majority of users of these transforms will not want stellar aberrations to be included, there are some edge cases where they would. While it is very unlikely that the coordinates of a star would be specified in ITRS coordinates, the ILRS does use ITRS ephemerides for some deep space objects where stellar aberration is required, an example being the Mars Global Surveyor.  This is why I included instructions for doing ITRS<->Observed transforms both with and without the effects of stellar aberration in my documentation of the ITRS frame.

Reply:
Thanks for the replies @mkbrewer. You've convinced me that this PR does the "right thing". Hopefully my suggested doc tweaks above will make it clearer when to use `transform_to` which treats the coordinate as a deep space object, and when to use the approach outlined in the `itrs.py` file for the less expert user.